llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -1345,18 +1350,30 @@ AArch64TargetLowering::AArch64TargetLowering(const TargetMachine &TM,
     }
 
     // AArch64 has implementations of a lot of rounding-like FP operations.
+    // And the same for FMAXNUM_IEEE and FMINNUM_IEEE.
     for (auto Op :
          {ISD::FFLOOR, ISD::FNEARBYINT, ISD::FCEIL, ISD::FRINT, ISD::FTRUNC,
-          ISD::FROUND, ISD::FROUNDEVEN, ISD::STRICT_FFLOOR,
-          ISD::STRICT_FNEARBYINT, ISD::STRICT_FCEIL, ISD::STRICT_FRINT,
-          ISD::STRICT_FTRUNC, ISD::STRICT_FROUND, ISD::STRICT_FROUNDEVEN}) {
+          ISD::FROUND, ISD::FROUNDEVEN, ISD::FMAXNUM_IEEE, ISD::FMINNUM_IEEE,
+          ISD::STRICT_FFLOOR, ISD::STRICT_FNEARBYINT, ISD::STRICT_FCEIL,
+          ISD::STRICT_FRINT, ISD::STRICT_FTRUNC, ISD::STRICT_FROUND,
+          ISD::STRICT_FROUNDEVEN}) {
       for (MVT Ty : {MVT::v2f32, MVT::v4f32, MVT::v2f64})
         setOperationAction(Op, Ty, Legal);
       if (Subtarget->hasFullFP16())
         for (MVT Ty : {MVT::v4f16, MVT::v8f16})
           setOperationAction(Op, Ty, Legal);
     }
 
+    // In fact TargetLowering::expandFMINIMUMNUM_FMAXIMUMNUM works well with
+    // them. While in narrowInsertExtractVectorBinOp, they are expected to be
+    // LegalOrCustom.
```

Comment:
This comment is too specific and about details of other code. I'm not sure what "works well with them" means, just delete the whole thing?

Reply:
This is just working around missing vector legalization support

Reply:
Sorry. I don't understand it well.
The IR here is about 256bit operation, while NEON has 128bit support, thus a split is needed anyway.
Do you mean that we need to split it as an ISD operation?

Reply:
The default vector legalization should give you the splitting result, this doesn't require custom lowering to handle

Reply:
#109779 handles the widening case. You need to add the FMINIMUMNUM/FMAXIMUMNUM cases to DAGTypeLegalizer::SplitVectorResult, like is done for FMINIMUM/FMAXIMUM already

Reply:
Coolll, thanks.